chunky/src/java/se/llbit/chunky/renderer/scene/WaterShading.java

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2012-2021 Chunky contributors
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.renderer.scene;
+
+import se.llbit.chunky.model.WaterModel;
+import se.llbit.math.Ray;
+import se.llbit.math.SimplexNoise;
+import se.llbit.math.Vector3;
+
+public class WaterShading {
```

Comment:
Maybe we can extract an interface from this to allow plugins to hook into water shading? And then have two implementations (`SimplexWaterShader implements WaterShader` and `LegacyWaterShader implements WaterShader`)?

Reply:
And doing the usual registry of implementations, dropdown instead of checkbox and the work I suppose? Why not

Reply:
Maybe one day, but that's not blocking this PR from being merged. :wink: For now, I'd prefer interface+inheritance instead of the `useFixedTexture` magic boolean.